### PR TITLE
Fix pod stuck in termination state when mount fails or gets skipped after kubelet restart

### PR DESCRIPTION
**Pull request by gnufied**
This is a non-automated cherry-pick of https://github.com/kubernetes/kubernetes/pull/110670

/sig storage
/sig node

### CODE REVIEW

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -75,21 +75,31 @@ type Reconciler interface {
 // NewReconciler returns a new instance of Reconciler.
 //
 // controllerAttachDetachEnabled - if true, indicates that the attach/detach
-//   controller is responsible for managing the attach/detach operations for
-//   this node, and therefore the volume manager should not
+//
+//	controller is responsible for managing the attach/detach operations for
+//	this node, and therefore the volume manager should not
```
Comment (jsafrane) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:80]: Please fix the format, i.e. only a single space after `//` and both 1.18 and 1.19 gofmt should be the same. See also below for other glitches in this file.

**Review by jsafrane — COMMENTED** [review]
almost lgtm, except for gofmt.

File: pkg/kubelet/volumemanager/reconciler/reconciler.go (hunk 1 above)
Comment (gnufied) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go:80]: I am not sure I understand this completely. I tried running `gofmt` manually against this file and it produced no change (My editor is using `lsp-format` action)

Comment (jsafrane) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go:80]: go 1.19 reformats `//  foo` (with two spaces) into `//      foo`. It expects a single space after `//`

Comment (gnufied) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go:80]: Understood. Although I wonder why isn't running `gofmt` manually in my local environment producing that diff.

Comment (jsafrane) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go:80]: `¯\_(ツ)_/¯`

Comment (gnufied) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go]: I have pushed a "better" format. You are right in the sense that yes `gofmt` insists on leading line in a comment to have one space between `//` and `text`. 

But - when you are inside the comment block and you indent a text more than one space, it tries to indent it by a tab..

**Comment by gnufied** [comment]
cc @kubernetes/release-managers

**Review by saschagrunert — APPROVED** [review]
Do we only require backports for 1.23?

**Pull request merged**